feat: add attendees frontmatter to notes and transcripts

kayacancode opened the pull request:
## Summary
Adds attendees information to the frontmatter of both synced notes and transcripts from Granola.

## Changes
- Extract attendees from `people.attendees` nested structure in Granola API response
- Add `attendees` field to frontmatter for both notes and transcripts
- Use lowercase `attendees:` field name (as per review feedback)
- Always include attendees when available from API (no settings toggle)

## Implementation Details
- Extracts attendee names from `doc.people.attendees` array
- Maps attendees to string array using `name` or `email` field (fallback to "Unknown" if neither exists)
- Filters out "Unknown" entries
- Formats attendees as YAML array in frontmatter:
  
  attendees:
    - John Doe
    - Jane Smith
  ## Files Changed
- `src/main.ts` - Added extraction logic in sync methods (daily notes, individual files, transcripts)
- `src/services/documentProcessor.ts` - Added attendees to note frontmatter
- `src/services/transcriptFormatter.ts` - Added attendees parameter and frontmatter support
- `src/services/granolaApi.ts` - Added `attendees` field to `GranolaDoc` interface
- `src/services/validationSchemas.ts` - Already had attendees validation schema
<img width="462" height="223" alt="Screenshot 2025-11-10 at 11 25 41 AM" src="https://github.com/user-attachments/assets/55f68410-8d63-440e-b72f-6c6c50882807" />

## Testing
- Attendees are extracted from API response and added to frontmatter when available
- Works for both notes and transcripts
- Only includes attendees when `people.attendees` exists in API response

kayacancode committed:
feat: add attendees frontmatter to notes and transcripts

- Extract attendees from people.attendees API structure
- Add attendees field to frontmatter for both notes and transcripts
- Use lowercase 'attendees' field name
- Always include when available from API (no settings toggle)

kayacancode commented:
Hey @tomelliot created a new PR. Let me know what you think.

tomelliot commented on src/services/granolaApi.ts:
```diff
@@ -25,6 +25,13 @@ export interface GranolaDoc {
   title: string | null;
   created_at?: string;
   updated_at?: string;
+  attendees?: string[];
```
@kayacancode do we get both `attendees` and `people.attendees` in the result from the API? I only see `people.attendees` in my results 🤔.

Let's keep the `GranoldaDoc` type as a subset of what we get from the API. That way we won't have to reorganise the data in the results in various places in the codebase - everything can just expect to get the same datatype as the API returns

kayacancode replied:
soudns good!

tomelliot commented on src/main.ts:
```diff
@@ -267,6 +267,15 @@ export default class GranolaSync extends Plugin {
   ): Promise<number> {
+    // Extract attendees from people.attendees for all documents before processing
+    for (const doc of documents) {
+      if (doc.people?.attendees && doc.people.attendees.length > 0) {
```
If we drop `attendees` from `GranolaDoc` and just use `people.attendees`, then we won't need to do this here (or with the transcripts)

tomelliot commented on src/main.ts:
```diff
@@ -347,7 +374,8 @@ export default class GranolaSync extends Plugin {
           doc.created_at,
-          doc.updated_at
+          doc.updated_at,
+          doc.attendees
```
We should be able to just pass in `doc.people.attendees`, without doing the map above.

tomelliot commented on src/services/transcriptFormatter.ts:
```diff
@@ -27,6 +29,11 @@ export function formatTranscriptBySpeaker(
   ];
   if (createdAt) frontmatterLines.push(`created_at: ${createdAt}`);
   if (updatedAt) frontmatterLines.push(`updated_at: ${updatedAt}`);
+  if (attendees && attendees.length > 0) {
```
@kayacancode what do you think about adding an empty array to the frontmatter, if there are no attendees (so that all frontmatter is consistent with the same keys)

kayacancode replied:
Hey @tomelliot thats a good idea. we should just always include it even if it's empty. Makes everything more consistent also.

kayacancode replied:
<img width="467" height="127" alt="Screenshot 2025-11-10 at 8 22 09 PM" src="https://github.com/user-attachments/assets/f9dc9c1b-7c4c-4ef5-9dba-68d5f2e94639" />

lfg!

tomelliot reviewed (changes requested):
Thanks @kayacancode this is great - a couple of comments but then it's pretty much ready to merge

kayacancode committed:
refactor: use people.attendees directly and always include attendees field

- Remove intermediate doc.attendees field
- Extract attendees inline from doc.people.attendees where needed
- Always include attendees field in frontmatter (empty array when no attendees)
- Remove attendees from validation schema (only validate people.attendees structure)

kayacancode commented:
Hey @tomelliot , just pushed those changes. Let me know what you think!

tomelliot commented:
Thanks @kayacancode !
